code/datums/components/traits/waddle.dm

```diff
@@ -0,0 +1,103 @@
+/datum/component/waddle_trait
+	var/atom/movable/our_atom
+	var/mob/living/living_owner
+
+	var/waddling = 1
+	var/waddle_z = 4
+	var/waddle_min = -12
+	var/waddle_max = 12
+	var/waddle_time = 2
+
+
+/datum/component/waddle_trait/Initialize()
+	if (!isobj(parent) && !ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(isliving(parent))
+		living_owner = parent
+		add_verb(living_owner, /mob/living/proc/waddle_adjust)
+		//add_verb(living_owner, /mob/living/proc/waddle_debug)
+	our_atom = parent
+	RegisterSignal(our_atom, COMSIG_MOVABLE_MOVED, PROC_REF(process))
+
+/datum/component/waddle_trait/process()
```

Comment:
this should probably not be process since it's not actually being processed, but rather a proc

Reply:
ye, I had thought about renaming all the traits (all three of the trait PRs currently use process) from using process() to another proc name...I think I'll just rename them to 'comp_activation' for simplicity's sake.